### PR TITLE
177 RST stubs for ReadTheDocs

### PR DESCRIPTION
**Pull request by bbean23**
## Purpose

_Please briefly describe the purpose of this pull request. If this fixes a github issue please include `e.g. fixes #22` so that the issue gets closed automatically when this is merged._

## Summary of changes

- provided RST stubs for SpotAnalysis classes
- updated docstrings
- added classes to test_DocStringsExist.py

## Implementation notes

Added AutoSummary to Sphinx for generating class name lists at the top of the documentation files.

## Submission checklist
- [x] Target branch is `develop`, not `main`
- [x] Existing tests are updated or new tests were added
- [x] `opencsp/test/test_DocStringsExist.py` are verified to include this change or have been updated accordingly
- [x] .rst file(s) under `doc/` are verified to include this change or have been updated accordingly

## Additional information

none

### CODE REVIEW

File: opencsp/common/lib/cv/spot_analysis/SpotAnalysisOperablesStream.py
Comment (e10harvey) [inline opencsp/common/lib/cv/spot_analysis/SpotAnalysisOperablesStream.py:37]: Missing parameter documentation in doc string

File: doc/source/library_reference/library/SpotAnalysis/base_classes.rst
Comment (e10harvey) [inline doc/source/library_reference/library/SpotAnalysis/base_classes.rst:25]:
```suggestion

.. automodule:: opencsp.common.lib.cv.spot_analysis.SpotAnalysisOperablesStream
   :members:
   :undoc-members:
   :show-inheritance:
   :member-order: groupwise

```

File: opencsp/common/lib/cv/spot_analysis/image_processor/FalseColorImageProcessor.py
Comment (e10harvey) [inline opencsp/common/lib/cv/spot_analysis/image_processor/FalseColorImageProcessor.py:39]: Missing parameter docs for operable

File: doc/source/library_reference/library/SpotAnalysis/image_processor.rst
Comment (e10harvey) [inline doc/source/library_reference/library/SpotAnalysis/image_processor.rst:16]:
```suggestion
   AbstractSpotAnalysisImageProcessorLeger
   AbstractVisualizationImageProcessor
```

File: opencsp/test/test_DocStringsExist.py
Comment (e10harvey) [inline opencsp/test/test_DocStringsExist.py:65]: thank you

**Review by e10harvey — CHANGES_REQUESTED** [review]
Thank you for these changes @bbean23. I left some minor comments below. Please reorganize these rst files to match the directory structore of the code base, like this:
`doc/source/library_reference/library/SpotAnalysis -> doc/source/library_reference/common/lib/cv/spot_analysis`. I'd be happy to open a PR into your fork with these changes.

**Review by bbean23 — COMMENTED** [review]
I believe I've addressed all your comments. Please note that by "addressed" I sometimes mean "argued against", so please take a look at my comments to see if you agree or not.

File: doc/source/library_reference/library/SpotAnalysis/base_classes.rst
Comment (e10harvey) [reply doc/source/library_reference/library/SpotAnalysis/base_classes.rst]: I don't have a strong preference but the current documentation uses the second style

File: doc/source/library_reference/library/SpotAnalysis/index.rst
Comment (e10harvey) [reply doc/source/library_reference/library/SpotAnalysis/index.rst]: Either way is fine with me but I want to make sure both the code and rst is maintainable and organized. Maybe the compromise is appending `config_` to base_classes.rst and image_processor.rst.

File: doc/source/library_reference/library/SpotAnalysis/base_classes.rst
Comment (e10harvey) [reply doc/source/library_reference/library/SpotAnalysis/base_classes.rst]: I would be inclined to include all non-protected and non-private entities in the documentation

**Review by e10harvey — COMMENTED** [review]
Thank you @bbean23! I think this can be merged after the headings are made consistent with the rest of the library reference.

File: doc/source/library_reference/library/SpotAnalysis/index.rst
Comment (bbean23) [reply doc/source/library_reference/library/SpotAnalysis/index.rst]: I'm not sure what you mean by this. You mean changing the file name to `base_classes_config.rst` and `image_processor_config.rst`?

File: opencsp/common/lib/cv/SpotAnalysis.py
Comment (e10harvey) [inline opencsp/common/lib/cv/SpotAnalysis.py:47]:
```suggestion
        i. Earth mover's metric for comparisons (scipy has a method, also look at https://www.cs.cmu.edu/~efros/courses/LBMV07/Papers/rubner-jcviu-00.pdf)
```

**Review by e10harvey — APPROVED** [review]

File: doc/source/library_reference/library/SpotAnalysis/index.rst
Comment (bbean23) [reply doc/source/library_reference/library/SpotAnalysis/index.rst]: To be addressed in follow-on issue

File: doc/source/library_reference/library/SpotAnalysis/base_classes.rst
Comment (bbean23) [reply doc/source/library_reference/library/SpotAnalysis/base_classes.rst]: Todo: indicate that this class is not meant for use by users, and most of the time not for use by developers. It is necessary for how the SpotAnalysis class works but only users who really want a deep dive should take a look. Maybe _SpotAnalysisOperablesStream or __SpotAnalysisOperablesStream?

Comment (bbean23) [reply doc/source/library_reference/library/SpotAnalysis/base_classes.rst]: Done!

File: opencsp/common/lib/cv/SpotAnalysis.py
Comment (bbean23) [reply opencsp/common/lib/cv/SpotAnalysis.py]: Thanks for your fix on this @e10harvey !

**Comment by bbean23** [comment]
@e10harvey Please review again. I made some small changes in order to get the build to pass and documentation to render correctly:

* fixed docstrings in CacheableImage to be compatible with ReStructuredText
* changed "AbstractSpotAnalysisImagesProcessor" to "AbstractSpotAnalysisImageProcessor"

**Pull request merged**
